### PR TITLE
feat(dashboard): add consolelink for ODH

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

ref: https://issues.redhat.com/browse/RHOAIENG-3760 

## How Has This Been Tested?
**NEW**  
new live build after rebase quay.io/wenzhou/opendatahub-operator-catalog:v2.9.884-2
`this build is using main branch from ODH odh-dashboard, since the old incubation is not updated so the consolelink.yaml not even exist there`
on ODH
Screenshot from 2024-03-08 10-00-34

**OLD**
live-build:  quay.io/wenzhou/opendatahub-operator-catalog:v2.9.884-1

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/dashboard/dashboard.go [hunk 1]
```diff
@@ -240,7 +250,7 @@ func (d *Dashboard) deployConsoleLink(cli client.Client, owner metav1.Object, pl
 	domainIndex := strings.Index(consoleRoute.Spec.Host, ".")
 	consoleLinkDomain := consoleRoute.Spec.Host[domainIndex+1:]
 	err := common.ReplaceStringsInFile(pathConsoleLink, map[string]string{
-		"<dashboard-url>": "https://rhods-dashboard-" + namespace + "." + consoleLinkDomain,
+		"<dashboard-url>": "https://" + componentName + "-" + namespace + "." + consoleLinkDomain,
```
Comment (lucferbux) [inline components/dashboard/dashboard.go:253]: I've just tested this in my cluster, I lack the whole context so I'm not sure where componentName comes from, but in ODH, the formatted url is: `https://dashboard-opendatahub.apps...` when the actual url is ``https://odh-dashboard-opendatahub.apps...`, it's missing the odh prefix here.

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: components/dashboard/dashboard.go (hunk 1 above)
Comment (zdtsw) [reply components/dashboard/dashboard.go:253]: need to know how you create this URL , is it hardcoded by dashboard to always set to https://odh-dashboard-opendatahub.* for ODH, or if that might change in different cases

Comment (zdtsw) [reply components/dashboard/dashboard.go]: updated the code to hardcode ODH for `odh-dashboard`

**Review by lucferbux — APPROVED** [review]
/lgtm

I think the code looks great now, but it would be nice if you can pass another live build to test the change and verify it's working

**Comment by zdtsw** [comment]
> /lgtm
> 
> I think the code looks great now, but it would be nice if you can pass another live build to test the change and verify it's working

i have re-installed quay.io/wenzhou/opendatahub-operator-catalog:v2.9.884-2 and it is pointing to the url which can open dashboard

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**